### ISSUE TITLE
[Question] How can I use fluid_event_timer?

### ISSUE THREAD

**Issue by sunny-lan** [atsushieno/nfluidsynth#10]
The FluidSynth documentation says fluid_event_timer can be used to set a arbitrary callback at a MIDI tick.

Is this implemented in this binding? How can I use it?

**Comment by atsushieno** [comment]
Use `SequencerEvent.Timer()` ? https://github.com/atsushieno/nfluidsynth/blob/fa077e0ed1081e3e7aba0a48dfbc4f17efd19e8f/NFluidsynth/SequencerEvent.cs#L166

**Comment by atsushieno** [comment]
I'm not familiar with either the native API or the managed API (it's not what I wrote), but looks like you should be able to create `Sequencer` instance, set `SequencerEventCallback` using `RegisterClient`.

https://github.com/atsushieno/nfluidsynth/blob/fa077e0ed1081e3e7aba0a48dfbc4f17efd19e8f/NFluidsynth/Sequencer.cs#L68

**Comment by sunny-lan** [comment]
Ok, thanks, I figured it out. I'm not great with PInvoke but is there a reason Timer takes void* data rather than IntPtr?

**Comment by atsushieno** [comment]
Interesting, that's a good catch. I don't see any particular reason for that. Unless @PJB3005 has any we can change that. On the other hand I don't also see particular reason to dare to bring in such public API breakage though - it's just one cast away.

**Comment by PJB3005** [comment]
It's kind of a personal taste but the reason I use `void*` is because it communicates intent more clearly than `IntPtr`, which is a good thing if you're gonna be messing with pointers in the first place IMO. Also it matches the Fluidsynth C API 1:1 which is always good.

I do see the nuisance of having to enable unsafe code in your project if you're just gonna pass a `0` or index or such though so it may be a good idea to provide an overload that's an `IntPtr`.

**Comment by sunny-lan** [comment]
Yeah, I agree with both of your points. An overload would be great as I just wanted to pass 0.
